### PR TITLE
#4219 emit sound system add single sound support

### LINKED ISSUE #4219: EmitSoundSystem needs to be able to take both soundCollections and sounds
## Description

Currently Vox/Snake/Mouse plushies make no sound when squeezed because whoever refactored ToysComponent into EmitSound component forgot to update the `sound:` fields in the yaml. 

I don't think creating a sound collection for single sounds is good practice so EmiteSoundSystem needs to take both soundCollection IDs and sound paths.

### PR DESCRIPTION
**Pull request by GalacticChimp**
Fixes #4219

### CODE REVIEW

**Comment by GalacticChimp** [comment]
Calling @SweptWasTaken and @RemieRichards  since they opened the issue

**Review by metalgearsloth — APPROVED** [review]
Also check it doesn't give log errors in-game too.

**Comment by metalgearsloth** [comment]
Ultimately this should be replaced with SoundSpecifier whenever we get that some day (like SpriteSpecifier... but for sounds).

**Comment by metalgearsloth** [comment]
Okay so zumzum is now working on it so maybe revisit after #4239 is merged if you want the ideal version.

**Comment by GalacticChimp** [comment]
@metalgearsloth ok, I'll wait a bit and do a proper full version, so we won't have to get back to this

**Comment by GalacticChimp** [comment]
Don't merge yet, I'll add SoundSpecifier in other parts of the code as well

**Comment by Zumorica** [comment]
Wouldn't it be better if you did multiple PRs? They'd be easier to review

**Comment by GalacticChimp** [comment]
> Wouldn't it be better if you did multiple PRs? They'd be easier to review

on second thought, yes, this is a good idea

File: Content.Server/Sound/EmitSoundSystem.cs [hunk 1]
```diff
@@ -1,55 +1,50 @@
...
         {
             base.Initialize();
-            SubscribeLocalEvent<EmitSoundOnLandComponent, LandEvent>((eUI, comp, arg) => PlaySound(comp));
-            SubscribeLocalEvent<EmitSoundOnUseComponent, UseInHandEvent>((eUI, comp, arg) => PlaySound(comp));
-            SubscribeLocalEvent<EmitSoundOnThrowComponent, ThrownEvent>((eUI, comp, arg) => PlaySound(comp));
-            SubscribeLocalEvent<EmitSoundOnActivateComponent, ActivateInWorldEvent>((eUI, comp, args) => PlaySound(comp));
+            SubscribeLocalEvent<EmitSoundOnLandComponent, LandEvent>((eUI, comp, arg) => HandleEmitSoundOn(comp));
+            SubscribeLocalEvent<EmitSoundOnUseComponent, UseInHandEvent>((eUI, comp, arg) => HandleEmitSoundOn(comp));
+            SubscribeLocalEvent<EmitSoundOnThrowComponent, ThrownEvent>((eUI, comp, arg) => HandleEmitSoundOn(comp));
+            SubscribeLocalEvent<EmitSoundOnActivateComponent, ActivateInWorldEvent>((eUI, comp, args) => HandleEmitSoundOn(comp));
```
Comment (Zumorica) [inline Content.Server/Sound/EmitSoundSystem.cs:28]: This is a bad idea, just make your methods take in the three arguments, even if two of them go unused.

**Review by Zumorica — CHANGES_REQUESTED** [review]

File: Content.Server/Sound/EmitSoundSystem.cs [hunk 2]
```diff
@@ -1,55 +1,63 @@
...
-            return _random.Pick(soundCollection.PickFiles);
+            TryEmitSound(component);
         }
 
-        private static void PlaySingleSound(string soundName, BaseEmitSoundComponent component)
+        private static void TryEmitSound(BaseEmitSoundComponent component)
         {
-            SoundSystem.Play(Filter.Pvs(component.Owner), soundName, component.Owner,
-                             AudioHelpers.WithVariation(component.PitchVariation).WithVolume(-2f));
+            if (!string.IsNullOrWhiteSpace(component.Sound.GetSound()))
```
Comment (GalacticChimp) [inline Content.Server/Sound/EmitSoundSystem.cs:53]: I'll add a TryGetSound() in a separate PR

**Pull request merged**